src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,100 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] array;
+    private int size;
+    private int threshold;
+
+    public MyHashMap() {
+        array = new Node[DEFAULT_CAPACITY];
+        threshold = (int) (DEFAULT_CAPACITY * LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size > threshold) {
+            resize();
+        }
+        int keyHashCode = getKeyHashCode(key);
+        int index = getIndexByKey(key);
+        Node<K, V> current;
+        current = array[index];
+        if (current != null) {
+            while (current != null) {
+                if (Objects.equals(current.key, key)) {
+                    current.value = value;
+                    return;
+                }
+                if (current.next == null) {
+                    current.next = new Node(keyHashCode, value, key, null);
+                    size++;
+                    return;
+                }
+                current = current.next;
+            }
+        }
+        array[index] = new Node(keyHashCode, value, key, null);
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        int index = getIndexByKey(key);
+        Node<K, V> current = array[index];
+        while (current != null) {
+            if (Objects.equals(current.key, key)) {
+                return current.value;
+            }
+            current = current.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void resize() {
+        int newLength = array.length * 2;
+        Node<K, V>[] oldArray = Arrays.copyOf(array, array.length);
+        array = new Node[newLength];
+        size = 0;
+        Node<K, V> current;
+        for (int i = 0; i < oldArray.length; i++) {
+            current = oldArray[i];
+            while (current != null) {
+                put(current.key, current.value);
+                current = current.next;
+            }
+        }
+        threshold = (int) (newLength * LOAD_FACTOR);
+    }
+
+    private int getKeyHashCode(K key) {
+        return key == null ? 0 : key.hashCode();
+    }
+
+    private int getIndexByKey(K key) {
+        return Math.abs(getKeyHashCode(key) % array.length);
+    }
```

Comment:
we can combine these two methods:
```suggestion
    private int getIndexByKey(K key) {
        return key == null ? 0 : Math.abs(key.hashCode()) % array.length;
    }
```

Reply:
fixed